### ISSUE TITLE
Use `torch-geometric` features and utils directly

### ISSUE THREAD

**Issue by arthurdjn** [arthurdjn/pytorch-pointcloud#73]
Instead of redefining some utils like `softmax`, `consecutive_cluster` etc., use the implementations provided by `torch-geometric` and make it a required dependency. Because we are re-using lots of utils from `torch-geometric` it could be convenient to reuse their implementations.

### Questions

- Should we reuse some conv? (`PointTransformerConv` and `PointNetConv` etc.)
- Should all our conv follow the `MessagePassing` interface?

-> Thing is, using `MessagePassing` interface creates a layer of abstraction. While I think it's fine, this layer of abstraction is poorly adopted in the 3D PointCloud community / research / papers. Most use more straight forward implementations
